src/rust/engine/src/downloads.rs

```diff
@@ -44,11 +46,21 @@ impl NetDownload {
   async fn start(
     core: &Arc<Core>,
     url: Url,
+    auth_headers: BTreeMap<String, String>,
     file_name: String,
   ) -> Result<NetDownload, StreamingError> {
+    let mut headers = HeaderMap::new();
+    for (k, v) in auth_headers.iter() {
+      headers.insert(
+        HeaderName::from_bytes(k.as_bytes()).unwrap(),
```

Comment:
Maybe avoid the `.unwrap` and return an error instead (using `?` operator to just do it inline)? I.e., `.map_err(|err| ...)?`.

Reply:
Same comment for the following line.

Reply:
In that case it'd be a dev mistake, so I'd rather crash-and-burn to make it obvious.